tests/tests.rs

```diff
@@ -429,6 +429,59 @@ async fn test_corrupted_index_regeneration() {
     warn!("elapsed: {:.3}", now.elapsed().as_secs_f64());
 }
 
+#[tokio::test]
+async fn test_corrupted_blob_index_regeneration() {
+    use std::mem::size_of;
+    use common::CorruptionType;
+
+    let now = Instant::now();
+    let path = common::init("corrupted_blob_index");
+    let storage = common::create_test_storage(&path, 70_000).await.unwrap();
+    let records = common::generate_records(100, 10_000);
+    for (i, data) in &records {
+        write_one(&storage, *i, data, None).await.unwrap();
+        sleep(Duration::from_millis(10)).await;
+    }
+    storage.close().await.unwrap();
+
+    let blob_header_size = 2 * size_of::<u64>() + size_of::<u32>();
+    let magic_len_size = size_of::<u64>() + size_of::<usize>();
+    let corruptions = [
+        // blob header corrupted
+        CorruptionType::ZeroedAtBegin(1),
+        // record header start corrupted
+        CorruptionType::ZeroedAt(blob_header_size as u64, 1),
+        // record header body corrupted
+        CorruptionType::ZeroedAt((blob_header_size + magic_len_size) as u64, 1)
+    ];
+    let len = corruptions.len();
+
+    for (i, corruption) in corruptions.into_iter().enumerate() {
+        let blob_file_path = path.join(format!("test.{i}.blob"));
+        let index_file_path = path.join(format!("test.{i}.index"));
+        assert!(blob_file_path.exists());
+        assert!(index_file_path.exists());
+
+        common::corrupt_file(blob_file_path, corruption).expect("blob corruption failed");
```

Comment:
This is a bad test, because several corruptions are applied at once in different places. This test would have passed even without changes, because blob header corruption worked properly before.
The proper way to implement this test is to create the storage, apply one corruption type, then verify, that the corruption was detected, and clean-up storage at the end.